### PR TITLE
[Feature/NifiCluster] Add abitility to set nodePort

### PR DESCRIPTION
**Pull request by juldrixx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/konpyutaika/nifikop/issues/283
| License         | Apache 2.0

### What's in this PR?

Adds the ability to set the NodePort Service nodePort. This field is new and optional, defaulting to nothing.

### Why?

By default and for convenience, the Kubernetes control plane will allocate a port from a range (default: 30000-32767) but some users want to specify it.

### Checklist

- [X] Implementation tested
- [X] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [X] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### CODE REVIEW

File: pkg/resources/nifi/service.go [hunk 1]
```diff
@@ -98,12 +99,17 @@ func (r *Reconciler) generateServicePortForExternalListeners(eService v1.Externa
...
-				usedPorts = append(usedPorts, corev1.ServicePort{
+				newPort := corev1.ServicePort{
 					Name:       strings.ReplaceAll(iListener.Name, "_", ""),
 					Port:       port.Port,
 					TargetPort: intstr.FromInt(int(iListener.ContainerPort)),
 					Protocol:   corev1.ProtocolTCP,
-				})
+				}
+
+				if eService.Spec.Type == corev1.ServiceTypeNodePort && port.NodePort != nil {
```
Comment (mh013370) [inline pkg/resources/nifi/service.go:109]: If the service type is node port and the user doesn't set a `PortConfig.NodePort`, then should we fail reconciliation with an error indicating this?

Does it make sense to configure a node port service, but then no node ports?

Comment (juldrixx) [reply pkg/resources/nifi/service.go:109]: This is to preserve the current behavior. If you set the service type to `NodePort`, the `nodePort` will be chosen randomly by the Kubernetes control plane (see https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport). I've just added the ability to specify this, but you can leave it out and have a random port as it is now.

Comment (mh013370) [reply pkg/resources/nifi/service.go:109]: Cool - sounds good!

File: api/v1/nificluster_types.go [hunk 2]
```diff
@@ -528,6 +528,9 @@ type PortConfig struct {
 	Port int32 `json:"port" protobuf:"varint,3,opt,name=port"`
 	// The name of the listener which will be used as target container.
 	InternalListenerName string `json:"internalListenerName"`
+	// The port that will expose this service externally. (Only if the service is of type NodePort)
+	// +optional
```
Comment (mh013370) [inline api/v1/nificluster_types.go:532]: Should we put some port validation on this (i.e. `0 <= nodePort <= 65535`)

**Review by mh013370 — APPROVED** [review]

**Pull request merged**